AudioSinkExample/ViewController.swift

```diff
@@ -0,0 +1,705 @@
+            logMessage(messageText: "Disconncted from \(room.name).\ncode = \((disconnectError as NSError).code) error = \(disconnectError.localizedDescription)")
+        } else {
+            logMessage(messageText: "Disconncted from \(room.name)")
+        }
+
+        for recorder in self.audioRecorders.values {
+            recorder.stopRecording()
+        }
+        self.audioRecorders.removeAll()
+
+        // Stop speech recognition!
+        stopRecognizingAudio()
+
+        self.room = nil
+
+        self.showRoomUI(inRoom: false)
+    }
+
+    func room(_ room: TVIRoom, didFailToConnectWithError error: Error) {
+        logMessage(messageText: "Failed to connect to Room:\n\(error.localizedDescription)")
+        self.room = nil
+
+        self.showRoomUI(inRoom: false)
+    }
+
+    func room(_ room: TVIRoom, participantDidConnect participant: TVIRemoteParticipant) {
+        participant.delegate = self
+
+        logMessage(messageText: "Participant \(participant.identity) connected with \(participant.remoteDataTracks.count) data, \(participant.remoteAudioTracks.count) audio and \(participant.remoteVideoTracks.count) video tracks")
+    }
+
+    func room(_ room: TVIRoom, participantDidDisconnect participant: TVIRemoteParticipant) {
+        logMessage(messageText: "Room \(room.name), Participant \(participant.identity) disconnected")
+    }
+}
+
+// MARK: TVIRemoteParticipantDelegate
+extension ViewController : TVIRemoteParticipantDelegate {
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           publishedVideoTrack publication: TVIRemoteVideoTrackPublication) {
+
+        // Remote Participant has offered to share the video Track.
+
+        logMessage(messageText: "Participant \(participant.identity) published \(publication.trackName) video track")
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           unpublishedVideoTrack publication: TVIRemoteVideoTrackPublication) {
+
+        // Remote Participant has stopped sharing the video Track.
+
+        logMessage(messageText: "Participant \(participant.identity) unpublished \(publication.trackName) video track")
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           publishedAudioTrack publication: TVIRemoteAudioTrackPublication) {
+
+        // Remote Participant has offered to share the audio Track.
+
+        logMessage(messageText: "Participant \(participant.identity) published \(publication.trackName) audio track")
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
+                           unpublishedAudioTrack publication: TVIRemoteAudioTrackPublication) {
+
+        // Remote Participant has stopped sharing the audio Track.
+
+        logMessage(messageText: "Participant \(participant.identity) unpublished \(publication.trackName) audio track")
+    }
+
+    func remoteParticipant(_ participant: TVIRemoteParticipant,
```

Comment:
Do we care about publish/unpublish events for data tracks in this example app?

Reply:
Nope, will remove.

Reply:
Removed.